src/test/scala/io/getclump/ClumpExecutionSpec.scala

```diff
@@ -150,6 +152,36 @@ class ClumpExecutionSpec extends Spec {
     promises.size mustEqual 2
```

Comment:
I've just noticed that this assertion is wrong, it always yields `true`. Could you fix the occurrences? My suggestion:

`promisesIterator.hasNext must beFalse`